simvue/run.py

```diff
@@ -37,81 +39,90 @@
-    Track simulation details based on token and URL
-    """
 
-    def __init__(self, mode="online"):
-        self._uuid = str(uuid.uuid4())
-        self._mode = mode
-        self._dispatch_mode = "queued"
-        self._name = None
+    @pydantic.validate_call
+    def __init__(self, mode: typing.Literal["online", "offline"] = "online") -> None:
+        """Initialise a new Simvue run
+
+        Parameters
+        ----------
+        mode : Literal['online', 'offline'], optional
+            mode of running, by default "online"
+        """
+        self._uuid: str = f"{uuid.uuid4()}"
+        self._mode: typing.Literal["online", "offline", "disabled"] = mode
+        self._name: typing.Optional[str] = None
+        self._dispatch_mode: typing.Literal["direct", "queued"] = "queued"
         self._executor = Executor(self)
-        self._dispatcher = None
-        self._id = None
-        self._suppress_errors = False
-        self._queue_blocking = False
-        self._status = None
-        self._upload_time_log = None
-        self._upload_time_event = None
-        self._data = {}
-        self._events = []
-        self._step = 0
-        self._queue_size = QUEUE_SIZE
-        self._metrics_queue = None
-        self._events_queue = None
-        self._active = False
+        self._dispatcher: typing.Optional[DispatcherBaseClass] = None
+        self._id: typing.Optional[str] = None
+        self._suppress_errors: bool = False
+        self._queue_blocking: bool = False
+        self._status: typing.Optional[
+            typing.Literal[
+                "created", "running", "completed", "failed", "terminated", "lost"
+            ]
+        ] = None
+        self._data: dict[str, typing.Any] = {}
+        self._step: int = 0
+        self._active: bool = False
         self._aborted: bool = False
         self._url, self._token = get_auth()
-        self._headers = {"Authorization": f"Bearer {self._token}"}
-        self._simvue = None
-        self._pid = 0
-        self._resources_metrics_interval = 30
-        self._shutdown_event = None
-        self._heartbeat_termination_trigger = None
-        self._storage_id = None
-        self._heartbeat_thread = None
-
-    def __enter__(self):
+        self._resources_metrics_interval: typing.Optional[int] = None
+        self._headers: dict[str, str] = {"Authorization": f"Bearer {self._token}"}
+        self._simvue: typing.Optional[SimvueBaseClass] = None
+        self._pid: typing.Optional[int] = 0
+        self._shutdown_event: typing.Optional[threading.Event] = None
+        self._heartbeat_termination_trigger: typing.Optional[threading.Event] = None
+        self._storage_id: typing.Optional[str] = None
+        self._heartbeat_thread: typing.Optional[threading.Thread] = None
+
+    def __enter__(self) -> "Run":
         return self
 
-    def __exit__(self, type, value, traceback):
+    def __exit__(
+        self, type: typing.Type[Exception], value: Exception, traceback: str
```

Comment:
Can we call this variable something other than `type`, which is a keyword, please?